src/main/java/org/mule/extension/http/api/HttpRequestAttributes.java

```diff
@@ -171,7 +178,21 @@ public String getRemoteAddress() {
   }
 
   public Certificate getClientCertificate() {
-    return lazyClientCertificate.get();
+    if (this.lazyClientCertificate != null && this.clientCertificate == null) {
+      this.clientCertificate = lazyClientCertificate.get();
+    }
+    return this.clientCertificate;
+  }
+
+  private void writeObject(ObjectOutputStream out) throws IOException {
+    //This will make the resolved client certificate be stored in the clientCertificate variable
+    getClientCertificate();
+    out.defaultWriteObject();
+  }
+
+  private void readObject(ObjectInputStream in) throws ClassNotFoundException, IOException {
```

Comment:
I don't think you need this. You are counting on the write setting the correct `clientCertificate` value, so the supplier is irrelevant since that has been resolved.